Stripe/STPPaymentContext.m

```diff
@@ -582,9 +582,10 @@ - (void)requestPayment {
         else if ([self requestPaymentShouldPresentShippingViewController]) {
             [self presentShippingViewControllerWithNewState:STPPaymentContextStateRequestingPayment];
         }
-        else if ([self.selectedPaymentMethod isKindOfClass:[STPCard class]]) {
+        else if ([self.selectedPaymentMethod isKindOfClass:[STPCard class]] ||
+                 [self.selectedPaymentMethod isKindOfClass:[STPSource class]]) {
```

Comment:
WDYT about checking `[self.selectedPaymentMethod conformsToProtocol:@protocol(STPSourceProtocol)]` instead?

I don't know: 
* whether there are other classes that conform to the protocol at we want excluded from this branch (making the check invalid)
* how likely it is that we'll add new classes that *do* conform to the protocol that should take this branch (updated check would future-proof us)
* how likely that we'll remove conformance to the protocol from these classes (which would break the check)

Reply:
mm, interesting idea. I think it's unlikely that we'll create other class that conform to this protocol (the protocol is mostly a compatibility shim for card tokens to work with sources). I think we should leave it this way and explicitly reference `STPCard`, which we should start treating as a legacy class (card tokens will continue to be supported for a long time, but sources are the preferred integration).